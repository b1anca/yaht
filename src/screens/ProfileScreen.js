import React, { useSelector } from "react-redux";

const ProfileScreen = () => {
  const { userInfo } = useSelector((state) => state.auth);

  return (
    <div>
      <div>profile</div>
      <span>
        Welcome <strong>{userInfo?.name}!</strong> You can view this page
        because you&apos;re logged in
      </span>
    </div>
  );
};

export default ProfileScreen;
