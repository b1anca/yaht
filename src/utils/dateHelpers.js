export const getDaysInMonth = (month, year) => {
  const date = new Date(year, month, 1);
  const days = [];
  while (date.getMonth() === month) {
    days.push(new Date(date));
    date.setDate(date.getDate() + 1);
  }
  return days;
};

export const formatDate = (date) => {
  const monthNames = [
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec",
  ];
  const day = date.getDate();
  const monthIndex = date.getMonth();
  const year = date.getFullYear();

  return day + " " + monthNames[monthIndex] + " " + year;
};

export const areDatesEqual = (date1, date2) => {
  return (
    date1.getFullYear() === date2.getFullYear() &&
    date1.getMonth() === date2.getMonth() &&
    date1.getDate() === date2.getDate()
  );
};
