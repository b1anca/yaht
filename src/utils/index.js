export const getRangeFromViewportWidth = () => {
  const viewportWidth = window.innerWidth;

  if (viewportWidth <= 390) {
    return 3;
  } else if (viewportWidth <= 500) {
    return 4;
    // sm
  } else if (viewportWidth <= 640) {
    return 6;
    // md
  } else if (viewportWidth <= 768) {
    return 8;
    // lg
  } else if (viewportWidth <= 1024) {
    return 9;
  } else {
    return 10;
  }
};
