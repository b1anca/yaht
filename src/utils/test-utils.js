import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { setupStore } from "../app/store";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return {
    store,
    user: userEvent.setup(),
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
