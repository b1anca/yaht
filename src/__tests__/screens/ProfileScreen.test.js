import React from "react";
import { screen, waitFor } from "@testing-library/react";
import { renderWithProviders } from "../../utils/test-utils";
import ProfileScreen from "../../screens/ProfileScreen";

describe("ProfileScreen", () => {
  it("should render correctly", async () => {
    const preloadedState = { auth: { userInfo: { name: "User name" } } };
    const { asFragment } = renderWithProviders(<ProfileScreen />, {
      preloadedState,
    });

    await waitFor(() => {
      screen.getByText(/user name/i);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
