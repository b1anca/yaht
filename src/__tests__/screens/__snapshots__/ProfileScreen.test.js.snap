// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen should render correctly 1`] = `
<DocumentFragment>
  <h2
    class="mb-4 font-extrabold tracking-tight text-zinc-600 dark:text-zinc-200 text-2xl"
    level="h4"
  >
    Your profile
  </h2>
  <p
    class="text-sm mb-2 tracking-tight text-zinc-600 dark:text-zinc-200"
  >
    Welcome 
    <strong>
      User name!
    </strong>
     You can view this page because you're logged in
  </p>
</DocumentFragment>
`;
